python/tvm/driver/tvmc/autotuner.py

```diff
@@ -139,6 +138,11 @@ def add_tune_parser(subparsers, _, json_params):
         help="enable tuning the graph through the AutoScheduler tuner",
         action="store_true",
     )
+    parser.add_argument(
+        "--visualize",
```

Comment:
this one reads like a boolean option (e.g. action="store_true"). could you better document the behavior?

Reply:
@areusch I also had the feeling that this is kinda unintuitive, since I added the option to write the visualizations directly to a file. How about introducing two separate command line flags `—visualize[-live]` (Type: `bool`) and `—visualize-[file|path]` (Type: `str`) to deal with that situation? If you would like that approach, please let me know how the two arguments should be called exactly.

Reply:
i think the existing arg is okay, i agree it's a flag with two meanings, but i think if you add it like this, the --help string will be clear:
```
NO_VISUALIZE_OPTION = object()
parser.add_argument("--visualize", nargs="?", type=str, metavar="file", default=NO_VISUALIZE_OPTION, help="Visualize tuning progress. With no parameter specified, display a pyplot window and update the graph as tuning progresses. With a parameter, instead write the graph to the file specified (format determined by the given file extension).")
```

(I think the "format determined by the given file extension" is correct, but maybe double check that)

Alternatively open to the two options, especially if you think users want both. you could also consider something like `--plot=live,foo.png`, which might provide a nice set of flexibility without users needing to know two options and how they interact.

Reply:
@areusch 
> `--plot=live,foo.png`

This should be what is implemented right now just with a different name. The main disadvantage here is that the custom format/parsing of the argument (i.e. `live, path.png`, `path.png,live`,...) can be confusing.

> `parser.add_argument("--visualize", nargs="?",   ...`

Your proposed solution which accepts either one or no args at all, looks nice at the first glance but the `nargs="?"` might lead to some unexpected behavior if a model path is confused with the argument to the `--visualize` option by argparse. In addition, as you already suggested this only allows:
- No visualizations
- Live only
- File only

That's why I would prefer the two separate options, which also allow a 4 combinations:
- No visualizations
- Live only
- File only
- Both